build.rs

```diff
@@ -230,13 +230,25 @@ fn parse_header_defines(header_path: impl AsRef<Path>) -> Result<HashMap<String,
     Ok(definitions)
 }
 
-fn fix_config_map(mut config_map: HashMap<String, String>) -> HashMap<String, String> {
-    if let Some("1") = config_map.get("Py_DEBUG").as_ref().map(|s| s.as_str()) {
+fn fix_config_map(
+    mut config_map: HashMap<String, String>,
+    version_minor: Option<u8>,
+) -> HashMap<String, String> {
+    // defining the Py_DEBUG macro no longer implies the Py_TRACE_REFS macro by py38
+    if version_minor.unwrap_or(0) >= 8 {
+        if let Some("1") = config_map.get("Py_DEBUG").as_ref().map(|s| s.as_str()) {
+            config_map.insert("Py_REF_DEBUG".to_owned(), "1".to_owned());
+        }
+    } else if let Some("1") = config_map.get("Py_DEBUG").as_ref().map(|s| s.as_str()) {
         config_map.insert("Py_REF_DEBUG".to_owned(), "1".to_owned());
         config_map.insert("Py_TRACE_REFS".to_owned(), "1".to_owned());
```

Comment:
I suggest this version with less repetition:

```suggestion
    if let Some("1") = config_map.get("Py_DEBUG").as_ref().map(|s| s.as_str()) {
        config_map.insert("Py_REF_DEBUG".to_owned(), "1".to_owned());
        if version_minor.map_or(false, |minor| minor <= 7) {
            // Py_DEBUG only implies Py_TRACE_REFS until Python 3.7
            config_map.insert("Py_TRACE_REFS".to_owned(), "1".to_owned());
        }
```

Reply:
Nice.